test/e2e_node/remote/ssh.go

```diff
@@ -70,7 +70,8 @@ func GetHostnameOrIP(hostname string) string {
 		host = ip
 	}
 
-	if *sshUser == "" {
+	// Do no ignore the KUBE_SSH_USER if it's set explicitly
+	if os.Getenv("KUBE_SSH_USER") != "" {
```

Comment:
I think the specified sshUser should be used instead of the env variable if specifying `--ssh-user` option.

Reply:
@oomichi Thanks for your reply. For the Fedora CoreOS host, the only user available on host is `core`.  We have seen failures like, 

```
prow@104.199.120.168: Permission denied (publickey,gssapi-keyex,gssapi-with-mic)
```
The `prow` user doesn't exist on the Fedora CoreOS. In fact, `KUBE_SSH_USER` was [introduced](https://github.com/kubernetes/kubernetes/pull/94049/files) to handle Fedora CoreOS hosts (which will be used extensively for CRI-O testing as well as in future RuntimeClass tests)

This PR doesn't prevent the existing flow, it's just that for Fedora CoreOS we want to make sure we use the username that we know works. 

cc @dims

Reply:
@harche Thanks for your response. I see your point.
I guess that comes from https://github.com/kubernetes/test-infra/blob/master/kubetest/e2e.go#L575 which gets the specified username on `--ssh-user` option from the env value `USER`.
I feel it is better to get the username from the env value `KUBE_SSH-USER` if existing. If not existing, get the username from the env value `USER` as current implementation.

I'd like to get help from @BenTheElder for the above idea.

Reply:
I agree that if --ssh-user is set we should respect that.
It also seems like https://github.com/kubernetes/test-infra/blob/feccbf60454db394d955e00977a9e2829712f050/kubetest/e2e.go#L575 could handle KUBE_SSH_USER overridding USER

Reply:
https://github.com/kubernetes/test-infra/pull/19273  making `kubetest/e2e.go` handle KUBE_SSH_USER overridding USER. 

cc @oomichi @BenTheElder